python/ray/serve/_private/application_state.py

```diff
@@ -368,7 +368,11 @@ def deploy_config(
                 ray.cancel(self._build_app_task_info.obj_ref)
 
             # Halt reconciliation of target deployments
-            self._set_target_state_deployment_infos(None)
+            self._set_target_state(
```

Comment:
Is it possible for this edge case to occur when the other calls to `self._set_target_state_deployment_infos(None)` happen? Specifically, [here](https://github.com/ray-project/ray/blob/c525af0b7e9cc89fffdb8ecea9f98c8a7b0c45bb/python/ray/serve/_private/application_state.py#L348) and [here](https://github.com/ray-project/ray/blob/c525af0b7e9cc89fffdb8ecea9f98c8a7b0c45bb/python/ray/serve/_private/application_state.py#L353)?

Reply:
Good catch, yeah that same edge case can happen if the config override options failed. Let me push an update to fix this too, thanks